ipv6/range.go

```diff
@@ -0,0 +1,93 @@
+package ipv6
+
+import (
+	"fmt"
+)
+
+// Range represents a range of addresses that don't have to be aligned to
+// powers of 2 like a Prefix does.
+//
+// Note that there is no instantiation of an empty range (.Size() == 0) because
+// .First() and .Last() would not make sense.
+//
+// The zero value of a Range is "[::, ::]" (.Size() == 1)
+type Range struct {
+	first, last Address
+}
+
+// NewRange returns a new range from the given first and last addresses. If
+// first > last, then empty is set to true and the returned range must be
+// ignored. There is no valid instantiation of an empty range.
+func NewRange(first, last Address) (r Range, empty bool) {
```

Comment:
What do you think about these ranges? To me, they are the strangest type in this library but I included them to be on par with other libraries I've seen like this (in go and other languages).

I cannot think of a good way to have an empty range (API-wise). I just can't. Do you have any thoughts?

Reply:
hmmm... The only thing I can really think of is to some way of having a range store a nil value. If I were in another language such as Java I would have an empty range be a null object. The only way I can think of is to return a pointer to a range struct. Nil would then represent an empty range.